### ISSUE TITLE
4.1.3 Rdlogedit: "log contains one or more carts that are invalid for the selected service"

### ISSUE THREAD

**Issue by paulk-2** [ElvishArtisan/rivendell#947]
4.1.3 running for weeks, when this suddenly started:

When opening a day's log in rdlogedit, it displays in a separate window:

    Invalid carts
    This log contains one or more carts that are invalid for the selected service!

There is only one service: "production".  There never has been another.

Another symptom appeared on one of the workstations:  in rdlogedit Service pulldown box, there were several services: all "production".  a reboot cleared that (back to one service).  Rebooting the main system did not clear the "invalid carts" error.

I captured the queries during the opening process, then did some manual queries, but did not find any clues yet.  Any pointers on what to look for in the database would be appreciated.

Has anyone else seen this error for a single-service system?

**Comment by KeepEmGoing** [comment]
Yes, nearly every time I open a log in rdlogedit it gives this message and I ignore it. I have not seen the replication of services that I recall. .

**Comment by paulk-2** [comment]
I have now run across an example where some tracks from a specific group were marked invalid (pink shading, not red) and they would not play.  Yet these carts were from the one and only production group.  After experimenting with new groups and moving carts from group to group, I got it to work again.  However I cannot replicate the exact process to get the carts marked as production group once again (in the GUI).   The database always had the group marked as Production.  There are no other services.

**Comment by paulk-2** [comment]
I found an extra entry in table RDLOGEDIT and deleted it.  That resolved the pink shading on carts belonging to a new group.  The "log contains carts... invalid for the selected service" message still appears, but now all carts can be edited ok.
Below is the table RDLOGEDIT before modification.  I deleted the last two lines and that cleared up the cart color.
The multiple entries for the STATION (AKA host) leads me to believe this is related to issue # 936, which has duplicate entries for the same host name.  I also found duplicate STATION names in tables RDLIBRARY, RDPANEL and DECKS.

select * from RDLOGEDIT;
+----+---------+------------+------------+-------------+-------------+--------+-------+---------+---------------------+------------------+-----------+--------------+------------------+------------+----------+----------------+--------------+----------------+--------------+--------------------+--------------+
| ID | STATION | INPUT_CARD | INPUT_PORT | OUTPUT_CARD | OUTPUT_PORT | FORMAT | LAYER | BITRATE | ENABLE_SECOND_START | DEFAULT_CHANNELS | MAXLENGTH | TAIL_PREROLL | WAVEFORM_CAPTION | START_CART | END_CART | REC_START_CART | REC_END_CART | TRIM_THRESHOLD | RIPPER_LEVEL | DEFAULT_TRANS_TYPE | IS_SINGLETON |
+----+---------+------------+------------+-------------+-------------+--------+-------+---------+---------------------+------------------+-----------+--------------+------------------+------------+----------+----------------+--------------+----------------+--------------+--------------------+--------------+
| 21 | W5      |          0 |          0 |           0 |           0 |      0 |     0 |       0 | Y                   |                2 |         0 |         2000 | %t - %a          |          0 |        0 |              0 |            0 |          -3000 |        -1300 |                  0 | Y            |
| 22 | w6      |          0 |          0 |           0 |           0 |      0 |     0 |       0 | Y                   |                2 |         0 |         2000 | %t - %a          |          0 |        0 |              0 |            0 |          -3000 |        -1300 |                  0 | Y            |
| 20 | W4      |          0 |          0 |           0 |           0 |      0 |     0 |       0 | Y                   |                2 |         0 |         2000 | %t - %a          |          0 |        0 |              0 |            0 |          -3000 |        -1300 |                  0 | Y            |
| 18 | z2      |          0 |          0 |           0 |           0 |      0 |     0 |       0 | Y                   |                2 |         0 |         2000 | %t - %a          |          0 |        0 |              0 |            0 |          -3000 |        -1300 |                  0 | Y            |
| 19 | z2      |         -1 |          0 |          -1 |           0 |      0 |     0 |       0 | Y                   |                2 |         0 |         2000 | %t - %a          |          0 |        0 |              0 |            0 |          -3000 |        -1300 |                  0 | Y            |

**Comment by ElvishArtisan** [comment]
The basic feature going on here is the ability to associate a Group with one or more Services. If a Group is not associated with a given Service, then it will not included in searches from the 'Add Cart' dialog in the on-air tools (RDAirPlay, RDPanel, etc). The warning and purple highlighs in RDLogEdit are merely an indicator that the given log contains carts that belong to a Group that is not listed as being one of the active services for that group. Those settings can be managed by editing the Group in question in RDAdmin and changing the 'Active Services' list.

> The multiple entries for the STATION (AKA host) leads me to believe this is related to issue # 936, which has duplicate entries for the same host name. I also found duplicate STATION names in tables RDLIBRARY, RDPANEL and DECKS.

I agree. There should be only one record per host ("STATION") in that table.

**Comment by paulk-2** [comment]
The original issue remains:

When opening a day's log in rdlogedit, it displays in a separate window:

   Invalid carts
   This log contains one or more carts that are invalid for the selected service!

There is only one service: "production". There never has been another.

All Groups have Production as an active service.

None of the carts are marked/colored as invalid.

Only the duplicate entry in RDLOGEDIT table was resolved.

**Comment by ElvishArtisan** [comment]
Fair enough.

Any chance of getting a copy of your DB?

**Comment by paulk-2** [comment]
I will try to create a trimmed down version that still triggers the behavior.
